src/Core/Form/ChoiceProvider/LanguageByIdChoiceProvider.php

```diff
@@ -52,11 +54,12 @@ public function __construct(LanguageDataProvider $languageDataProvider)
      *
      * @return array
      */
-    public function getChoices()
+    public function getChoices(array $options = [])
     {
         $choices = [];
+        $shopId = !empty($options['shop_id']) ? $options['shop_id'] : false;
```

Comment:
```suggestion
        $shopId = $options['shop_id'] ?? false;
```

Reply:
`??` is like doing an `'isset'`, if for some reason `$options['shop_id']` is set with let's say an empty string, it would be set as such, I prefer setting false as a value in all the cases where `empty` is true.

Reply:
Wrong : it checks that the value is not `false`

https://www.designcise.com/web/tutorial/whats-the-difference-between-null-coalescing-operator-and-ternary-operator-in-php

Reply:
`$shopId = $options['shop_id'] ?? false;` 

is equivalent to:

`$shopId = isset($options['shop_id']) ? $options['shop_id'] : false;`

It's actually written in your article :D

`expr1 ?? expr2;` is actually the same as the following:
`return (isset($x) ? $x : $y);`